draft-ietf-quic-spin-exp.md

```diff
@@ -235,12 +235,23 @@ Implementations SHOULD allow administrators of clients and servers to disable
 the spin bit either globally or on a per-connection basis.
 Even when the spin bit is not disabled by the administrator implementations
 SHOULD disable the spin bit on a randomly chosen
-fraction of connections.  The selection process should be designed such that
-on average the spin bit is disabled for at least 1/8th of the connections.
-
-When the spin bit is disabled, endpoints SHOULD set the spin bit value to zero,
-regardless of the values received from their peer. Addendums or revisions to
-this document MAY define alternative behaviors in the future.
+fraction of connections.
+
+The selection process SHOULD be designed such that
```

Comment:
One of the main concerns seems to be that an network could discriminate against a client that doesn't spin. For that reason, I think we should specify that an implementation that does spin should not spin for some percentage of clients. 

```suggestion
The selection process SHOULD be designed such that on average the spin
bit is disabled for at least one eighth of clients regardless of destination.
The selection process SHOULD be externally unpredictable but consistent for
any given client. For instance, the implementation might have a static key which
it uses to key a pseudorandom function over these values and use the output to 
determine whether to send the spin bit. The selection process performed for the first
connection on a given network and SHOULD be consistent for all subsequent 
connections on that network.
```

Reply:
I would prefer to keep this as a per-path option.  Having it per-client means that some people will never spin and can thus be identified by that as they move around.  It's a low entropy signal, but one that would add to linkability.

That said, I don't think that having the same client exhibit the same spin behaviour on a given path when time passes is right either.  If implemented poorly, I could clear cookies and still be identifiable by whether I was spinning on a particular path or not.

Based on this, I'm going to advocate for dropping the consistency recommendation as well.  Can someone restate the problem we were attempting to solve with that?

Reply:
This is the "server behind a proxy" problem. Clients connecting to that server from a dangerous place will most likely disable spinning for all connections to that server -- in fact, the server itself might do that. So you get a client/server address pair for which you never see spinning. You don't want that to stick out, so you need to ensure that there are other client/server address pairs that never spin.

Reply:
It's more than that.  If I consistently spin or not for a given path then I leak some bits of entropy about whether I was the same entity last time I used that path.  And I leak that to everyone on the network.  Consistency => linkability.

Reply:
By the way, just like Martin, I do not think that we should make this per-client. In fact, I don't think this is practical. Consider a device running Firefox, Google Chrome, and the Facebook app. Three different applications, each with their own stack, each managed their own way. You would have to count them as 3 clients. But then, their connections share the same IP source address. So what would disabling 1/8th of those achieve, exactly?

Reply:
@martinthomson the text mentions making the decision based on the addresses for the path -- client address and server address. If the client changes location, it will change address, and the decision will not be correlated to what happens at different locations.

Reply:
But unless the algorithm is specified, each of those implementations might choose differently.  Doesn't that affect things as well?

Reply:
I sense we might be over engineering this...

Reply:
I think we have consensus on that.

Reply:
That was my point.  In addition to the recommendation about stable decisions being over-engineering, I believe that it's a privacy risk, because it lets an observer link activity from the same location over time.  It's not a huge risk, but when we're so specific about advice, people will implement that and when that advice is potentially bad, that's not great.

Reply:
Fingerprinting might be easier to track than stable decisions.

Reply:
Agreed that this might be getting to prescriptive. Perhaps its best to call out the concerns to be addressed by implementers and leave it at that.

```suggestion
The selection process SHOULD be designed such that on average
the spin bit is disabled for at least one eighth of connections. The implementation
SHOULD take linkability across sessions where state would otherwise be cleared as well as
masking whether the client probabilistically spins or never spins into
account when designing the selection process.
```